index.js

```diff
@@ -1,88 +1,125 @@
+var counter = 1
+console.log("%cQuestion 1", "color: red") 
+console.log(counter) 
 
-// console.log(counter) 
-// // => 2
-// console.log("%c----------", "color: red") 
+
+// => 2
+var counter = 2
+console.log(counter)
+console.log("%c----------", "color: red") 
 
 
 // ***** Question 2 *****
 // *** Uncomment the lines below to test
 // *** after testing, comment the line causing the error back in 
 // *** otherwise, the error will stop the rest of your code from running
-// console.log("%cQuestion 2", "color: red") 
+console.log("%cQuestion 2", "color: red") 
+
 
-// console.log(name)
+const name = 'Henry Velasquez'
+console.log(name)   // WHY DOES NAME HAVE A STRIKETHROUGH?
 // => "Raffy"
-// name = "Not Raffy" 
+
 // => TypeError
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 3 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
+console.log("%cQuestion 3", "color: red") 
 
-// console.log(drinkWater(12))
+function drinkWater(currentThirstLevel) { 
+  console.log("Man I sure am thirsty")
+  currentThirstLevel -= 1
+  console.log( 'Ahh that hits the spot' )
+  return currentThirstLevel
+}
+ console.log(drinkWater(12))
 // // => "Man I sure am thirsty"
 // // => "Ahh that hits the spot"
 // // => 11
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
+console.log("%cQuestion 4", "color: red") 
+
+function sameSameButDifferent(num, maybeNum) {
+  if (num === maybeNum) {
+    console.log('same same')
+  } else if (num === parseInt(maybeNum, 10)) {
+    console.log( 'same same but different' ) //DO I PUT SPACES OR NO?
+  } else {
+    console.log('different')
+  }
+}
 
-// console.log(sameSameButDifferent(5, 5)) 
-// // => "same same"
+console.log(sameSameButDifferent(5, 5)) 
+// => "same same"
 
-// console.log(sameSameButDifferent(123, "123")) 
-// // => "same same (but different)"
+console.log(sameSameButDifferent(123, "123")) 
+// => "same same (but different)"
 
-// console.log(sameSameButDifferent(5, 7)) 
-// // => "different"
+console.log(sameSameButDifferent(5, 7)) 
+// => "different"
 
-// console.log(sameSameButDifferent(123, "122")) 
-// // => "different"
-// console.log("%c----------", "color: red") 
+console.log(sameSameButDifferent(123, "122")) 
+// => "different"
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 5 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 5", "color: red")
-// const student1 = { name: "Duane", grade: 88 }
-// updateGrade(student1, 92)
-// console.log(student1)
+console.log("%cQuestion 5", "color: red")
+
+function updateGrade(student, grade) {
+  student['grade'] = grade   // WHAT EXACTLY IS THE CORRECT JSON OBJECT NOTATION. I TRIED [grade:] and [:grade] and they did not work
```

Comment:
Good! 👍  JSON stands for Javascript Object Notation! What you're doing here on line 82 is referred to as bracket notation (as opposed to dot notation) [you can see more here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Property_accessors)

You can also use dot notation like so:
`student.grade = grade`

The updates made to the `student` object inside the function is reflected in the `student1` object because objects are ["call by sharing" unlike primitive data types](https://stackoverflow.com/questions/518000/is-javascript-a-pass-by-reference-or-pass-by-value-language).